src/rtop/rtop.ml

```diff
@@ -1,32 +1,25 @@
-let getenv_opt env =
-  try Some(Sys.getenv env)
-  with Not_found -> None
+let () = UTop.require ["ocaml-migrate-parsetree"; "menhirLib";]
 
-(* Try to read the HOME environment
- * if it is not set we assume it's windows and fallback to USERPROFILE *)
-let home = match getenv_opt "HOME" with
-  | Some(p) -> p
-  | None -> Sys.getenv "USERPROFILE"
+let () = try Topdirs.dir_directory (Sys.getenv "OCAML_TOPLEVEL_PATH") with | Not_found -> ();;
 
-let () = Unix.openfile (Filename.concat home ".utoprc") [] 0o640
-         |> Unix.close
+let () = UTop.require ["reason.easy_format"; "reason";]
 
-let () = Unix.openfile (Filename.concat home ".utop-history") [] 0o640
```

Comment:
I wonder if it now wouldn't preserve history or load settings.

Reply:
It kept history when I tested while building it using `esy x rtop`, my guess is that it just loads `.utop-history` or something like that anyway.

Reply:
If it loads the same history as utop won't that cause issues on systems where both utop and rtop are used?

Reply:
I didn't get any history in rtop after using utop but I might have to test this some more

Reply:
Just tried again, can confirm that they do not share history. But it might be because esy sandboxes my rtop.

Reply:
We should make sure it doesn't fail on a clean install or at least figure out where it's getting the history from - it's getting it from *somewhere*. This was an original bug we had (the history file didn't exist) and rtop failed on first launch.

Reply:
https://github.com/facebook/reason/blob/master/src/rtop/reason_utop.ml#L54

This is where it comes from

Reply:
That's promising - it should continue to use an rtop-specific history file then.